lib/dialects/mysql/query.js

```diff
@@ -200,24 +202,26 @@ class Query extends AbstractQuery {
 
         return new sequelizeErrors.UniqueConstraintError({message, errors, parent: err, fields});
       }
-      case 1451: {
-        const match = err.message.match(/FOREIGN KEY \(`(.*)`\) REFERENCES `(.*)` \(`(.*)`\)(?: ON .*)?\)$/);
 
-        return new sequelizeErrors.ForeignKeyConstraintError({
-          fields: null,
-          index: match ? match[3] : undefined,
-          parent: err
-        });
-      }
+      case 1451:
       case 1452: {
-        const match = err.message.match(/FOREIGN KEY \(`(.*)`\) REFERENCES `(.*)` \(`(.*)`\)(.*)\)$/);
+        // e.g. CONSTRAINT `example_constraint_name` FOREIGN KEY (`example_id`) REFERENCES `examples` (`id`)
+        const match  = err.message.match(/CONSTRAINT `(.*)` FOREIGN KEY \(`(.*)`\) REFERENCES `(.*)` \(`(.*)`\)/);
+        const fields = match ? match[2].split(/`, *`/) : undefined;
+
+        // TODO: it is not clear whether `table` & `fields` is meant refer to the source or the referenced entity,
```

Comment:
We are exposing all fields given by error message, is there any way to get additional fields?

Reply:
@sushantdhiman - I do not understand your question! (how can there be "additional" things to "all" the things?)

*please note*: `fields` was an existing property, I assumed it was plural for a reason (my guess would be because of the possibility of compound foreign keys) - I chose to set `fields` as an array because of this.

Reply:
@iamjochem I am saying we are already exposing all fields available in error message. Your comment says we might want fields from source or referenced table. So I am asking is there any way to get those fields?

If not just remove that comment.

Reply:
aha - I see, comment has been removed. (the comment was mostly because I wasn't sure what "fields" was intended to contain - as per my longer reply/question below)